test/unit/spec/common/jobManager-Spec.js

```diff
@@ -819,150 +871,121 @@ define([
-                            the: 'other',
-                        },
-                        jobInfo = {
-                            app_id: 'some/app',
-                            app_name: 'Some app',
-                            batch_id: null,
-                            job_id: this.jobId,
-                            job_params: [jobParams],
-                        };
-                    setUpHandlerTest(this, event);
-                    expect(
-                        this.jobManagerInstance.model.getItem(`exec.jobs.info.${jobId}`)
-                    ).not.toBeDefined();
+                Object.keys(inputs).forEach((inputName) => {
+                    const { channelType, channelId, message, jobIdList, error } = inputs[inputName],
+                        channelData = { channelType, channelId };
 
-                    return new Promise((resolve) => {
-                        // add a handler that acts after the default handler so we know
-                        // the default action has already been done
-                        this.jobManagerInstance.addListener(
-                            jcm.MESSAGE_TYPE.INFO,
-                            jcm.CHANNEL.JOB,
-                            jobId,
-                            {
+                    it(`handles successful job info requests, ${inputName}`, function () {
+                        setUpHandlerTest(this, event, channelType, [channelId]);
+                        jobIdList.forEach((jobId) => {
+                            expect(
+                                this.jobManagerInstance.model.getItem(`exec.jobs.info.${jobId}`)
+                            ).not.toBeDefined();
+                        });
+
+                        return new Promise((resolve) => {
+                            // add a handler that acts after the default handler so we know
+                            // the default action has already been done
+                            this.jobManagerInstance.addListener(event, channelType, channelId, {
                                 zzz_resolve_promise: (...args) => {
                                     // eslint-disable-next-line no-unused-vars
                                     const [_, msg, channel] = args;
-                                    expect(msg).toEqual(jobInfo);
+                                    expect(msg).toEqual(message);
                                     expect(channel).toEqual(channelData);
                                     resolve();
                                 },
+                            });
+                            TestUtil.sendBusMessage({
+                                bus: this.bus,
+                                message,
+                                channelType,
+                                channelId,
+                                type: event,
+                            });
+                        }).then(() => {
+                            jobIdList.forEach((jobId) => {
```

Comment:
different responses depending on whether the job manager has JOB listeners or a BATCH listener. If it has job listeners, it will remove the listener for the individual job, since the job info is now populated. If it has a batch listener, the listener will be left in place.